.github/workflows/build-test-linux.yml

```diff
@@ -261,3 +261,7 @@ jobs:
         cd tests/py/core
         ${CONDA_RUN} python -m pytest -n 4 --junitxml=${RUNNER_TEST_RESULTS_DIR}/tests_py_core_test_results.xml .
         popd
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.sha }}-${{ inputs.repository }}-${{ github.event_name == 'workflow_dispatch' }}-${{ inputs.job-name }}
+  cancel-in-progress: true
```

Comment:
This restriction may not be stringent enough, because I believe it allows multiple commits on the same branch to run CI, on account of the `${{ github.event.pull_request.number || github.sha }}` line. Consider using something similar to the concurrency group used for the Docker build:

https://github.com/pytorch/TensorRT/blob/87ba7f57b7cf3122363886c4edea38f2c9e6efab/.github/workflows/docker_builder.yml#L13-L15

Reply:
@gs-olive 
for the docker_builder.yml, I think concurency group: ${{ github.workflow }}-${{ github.ref_name }} is good for docker build workflow, however for the build-test-linux.yml or build-test-windows.yml, I still prefer the following: 
`${{ github.workflow }}-${{ github.event.pull_request.number || github.sha }}-${{ inputs.repository }}-${{ github.event_name == 'workflow_dispatch' }}-${{ inputs.job-name }}`

${{ inputs.job-name }} is required because this yml has many jobs such as test dynamo converters  or test dynamo frontend etc, we want those tests to be run in parallel.

${{ github.event.pull_request.number || github.sha }}-${{ inputs.repository }} is basically very similar to your github.ref_name
${{ github.event_name == 'workflow_dispatch' }} It depends on whether we want to cancel the build if someone is manually triggering a CI build from UI, it will use the latest github.sha at that time, in case there is a new sha committed in, we want to cancel it or not.
I prefer we keep this manual triggering not to be cancelled, because if I manually trigger a build on a specific sha, I want to be able to see the result of this sha, not want this got cancelled just because someone else has pushed in a new sha.

cc: @narendasan

Reply:
Presumably this is what we want, cancel on push to PR or branch 
```
${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}
```
https://stackoverflow.com/questions/66335225/how-to-cancel-previous-runs-in-the-pr-when-you-push-new-commitsupdate-the-curre

Reply:
Yes, I have changed
from 
${{ github.event.pull_request.number || github.sha }}
to 
${{ github.event.pull_request.number || github.ref_name }}